boot/bootutil/src/loader.c

```diff
@@ -1560,6 +1560,9 @@ boot_prepare_image_for_update(struct boot_loader_state *state,
          * if there is one.
          */
         BOOT_SWAP_TYPE(state) = BOOT_SWAP_TYPE_NONE;
+        (void) boot_read_image_header(state, BOOT_PRIMARY_SLOT,
```

Comment:
Don't know if there is any point in handling the return value of this. If it fails it fails and there is nothing we can do.

Reply:
Hmm, as I see the `boot_read_image_headers` should return with success if only the reading of the secondary slot has failed, if the `require_all` paramter set to `false`, but it seems to be okay when calling it in line 1570.

So, I would expect that execution is continued if only the reading of secondary slot has failed.